client/src/components/graphs/Pie.js

```diff
@@ -1,50 +1,38 @@
+import React, { useRef, useEffect } from "react"
+import PropTypes from "prop-types"
 import SVGCanvas from "components/graphs/SVGCanvas"
 import * as d3 from "d3"
-import PropTypes from "prop-types"
 
-export default class Pie extends SVGCanvas {
-  static propTypes = {
-    label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    segmentFill: PropTypes.func.isRequired,
-    segmentLabel: PropTypes.func.isRequired,
-    data: PropTypes.object.isRequired
-  }
+const Pie = props => {
+  const svgContainer = useRef(null)
+  const canvas = useRef(null)
+  const pie = useRef(d3.pie())
 
-  componentDidMount() {
-    this.canvas = this.svg
+  useEffect(() => {
+    canvas.current = d3
+      .select(svgContainer.current)
       .append("g")
-      .attr(
-        "transform",
-        `translate(${this.props.size.width / 2}, ${this.props.size.height / 2})`
-      )
-    this.pie = d3.pie()
-    this.pie.value(function(d) {
+      .attr("transform", `translate(${props.width / 2}, ${props.height / 2})`)
+    pie.current.value(function(d) {
       return d.value
     })
 
-    this.canvas
+    canvas.current
```

Comment:
`pie.current` could be `null` here?

Reply:
`pie` is reference initialized with a non-null value